### DISCUSSION TITLE
Listing and Searching Views

### DISCUSSION THREAD

**Discussion by ghislaineguerin** [mathesar-foundation/mathesar#669] [Maintainer discussions]
The design for issue #455 is ready for review. Please follow the [Review Guidelines](https://wiki.mathesar.org/en/design/process/review-guidelines) and make sure to review before the set deadline expires.

### Document
https://wiki.mathesar.org/en/design/specs/listing-views

### Reviewers
@kgodey @pavish @mathemancer  Your review is required.
Feedback from the community is also welcome. Contributors can find more context about Mathesar's design strategy [here](https://wiki.mathesar.org/design).

### Deadline
The deadline for submitting feedback is set for next week, on Tuesday 28th.

**Comment by kgodey** [comment]
This looks good to me. As always, we need to figure out how to make it work if there are hundreds of views, we can't display them all in the sidebar, even with scrolling.

Also, I like the idea of users being able to set default views, but it's going to be difficult for them to know which view is set to the default one already. Can we show this somehow?

I'm wondering whether we should have a "list of views" page, similar to our "list of schemas" page. We could also show which one is the default, which tables each reference, etc.

**Reply by ghislaineguerin** [reply]
@kgodey I had actually created a list of views within the same view as schemas but left it out from this design as I considered it was out of scope, but I agree we should. I will include it back. 
If we can't display all views, I wonder if this is also solved by that list of views, and what we show here is limited based on some criteria such as last modified. I will consider this scenario for the next round.

**Comment by pavish** [comment]
@ghislaineguerin I like the design. I like the idea of displaying views referencing a table. I do have some concerns:
1. I do not think we should allow setting a default view for the initial release. We should probably wait for all the features to be planned before we consider this.
   * We assume that users may want views to be open as soon as they open a schema, but we also discussed on opening previously opened views/tables when a schema is opened. We need to define how these would work together.
   * When forms, graphs, calendar and kanban apps etc., are implemented, how would setting a default view work? Wouldn't users want to set them as default too? How many items can a user set as default, and how do we decide priority among them?

2. We need to re-consider the left pane design (as discussed earlier). This is beyond scope of this discussion, but a reminder. 
    * Hiding/closing of left pane
    * We cannot list all views and tables in the left pane, since there can be large number of them
    * We need to consider how graphs, apps etc., would fit in the left pane
    * Searching when left pane is not open

**Reply by kgodey** [reply]
@pavish I think it would be best if you opened a new design issue for the left pane so that we can make sure to track the design work somewhere.

**Reply by pavish** [reply]
Created issue https://github.com/centerofci/mathesar/issues/723 for this.

**Comment by kgodey** [comment]
Based on the new action plan from the [2021-10-15 meeting](https://wiki.mathesar.org/meeting-notes/october-2021), we're going to update the requirements for issues in the milestone. I'm marking this discussion done, we'll start new discussions for further updates.
